src/partners/moonpay.ts

```diff
@@ -126,3 +146,41 @@ export function processMoonpayTx(rawTx: unknown): StandardTx {
   }
   return standardTx
 }
+
+function getFiatPaymentType(tx: MoonpayTx): FiatPaymentType | null {
+  switch (tx.paymentMethod) {
+    case undefined:
+      return null
+    case 'ach_bank_transfer':
+      return 'ach'
+    case 'apple_pay':
+      return 'applepay'
+    case 'credit_debit_card':
+      return 'credit'
+    case 'gbp_open_banking_payment':
+      return 'fasterpayments'
+    case 'google_pay':
+      return 'googlepay'
+    case 'mobile_wallet':
+      // Idk?
```

Comment:
Let's find out from moonpay what `mobile_wallet` is. possibly paypal or revolut. either way, we need to make sure all types are covered or we throw.

Reply:
I reached out to moonpay to find out. ⏳

Reply:
any resolution?

Reply:
Resolved. Looks like there is a cardType field in legacy documents. I'll leverage that to figure out if it's an applepay or googlepay exchangeType.